src/Aspire.Dashboard/Otlp/Model/OtlpApplication.cs

```diff
@@ -68,13 +70,17 @@ public void AddMetrics(AddContext context, RepeatedField<ScopeMetrics> scopeMetr
                             Context = Context
                         };
 
-                        instrument.AddMetrics(metric, ref tempAttributes);
+                        instrument = instrumentRef;
                     }
                     catch (Exception ex)
                     {
-                        context.FailureCount++;
-                        Context.Logger.LogInformation(ex, "Error adding metric.");
+                        // If we can't create the instrument then all data points for it are failures.
```

Comment:
What happens if this was the case that added the key to `_instruments` and then it failed? Does the entry need to get removed from the dictionary?

Reply:
An empty graph is displayed if the instrument was added but all its data points failed. I think that's fine.